docs/source/stability_policy.rst

```diff
@@ -0,0 +1,137 @@
+Deprecated attributes that are still passed by Telegram will be available in the :attr:`~telegram.TelegramObject.api_kwargs` dictionary as long as PTB can support that with feasible effort.
+Since attributes of the classes in the :mod:`telegram` package are not writable, we may change them to properties where appropriate.
+
+Development Versions
+~~~~~~~~~~~~~~~~~~~~
+
+Before a feature is in a release, it is not covered by this policy and may change.
+Pre-Release marked as alpha, beta or release candidate are not covered by this policy either.
+
+Security
+~~~~~~~~
+
+We make exceptions from our stability policy for security.
+We will violate this policy as necessary in order to resolve a security issue or harden PTB against a possible attack.
+
+Versioning
+----------
+
+PTB uses a versioning scheme that roughly follows `https://semver.org/ <https://semver.org/>`_, although it may not be quite as strict.
+
+Given a version of PTB X.Y.Z,
+
+-  X indicates the major version number.
+   This is incremented when backwards incompatible changes are introduced.
+-  Y indicates the minor version number.
+   This is incremented when new functionality or backwards compatible changes are introduced by PTB.
+   *This is also incremented when PTB adds support for a new Bot API version, which may include backwards incompatible changes in some cases.*
+-  Z is the patch version.
+   This is incremented for backwards incompatible changes or other major changes.
+
+Deprecation
+~~~~~~~~~~~
+
+From time to time we will want to change the behavior of an API or remove it entirely, or we do so to comply with changes in the Telegram Bot API.
+In those cases, we follow a deprecation schedule as detailed below.
+
+Functionality is marked as deprecated by a corresponding note in the release notes and the documentation.
+Were possible, a :class:`~telegram.warnings.PTBDeprecationWarning` is issued when deprecated functionality is used, but this is not mandatory.
+
+From time to time, we may decide to deprecate an API that is particularly widely used.
+In these cases, we may decide to provide an extended deprecation period, at our discretion.
+
+With version 20.0.0, PTB introduced majorly structural breaking changes without the above deprecation period.
+Should a similarly big change ever be deemed necessary by the development team and should a deprecation period prove too much additional effort, this violation of the stability policy will be announced well ahead of the release in our channel.
+
+.. _non-bot-api-functionality-1:
+
+Non-Bot API Functionality
+#########################
+
+Starting with version NEXT.VERSION, deprecated functionality will stay available for the current and the next major version.
+More precisely:
+
+-  In PTB X.Y.Z the feature exists
+-  In PTB X.Y.(Z + 1) or X.(Y + 1).\* the feature is marked as deprecated
+-  In PTB (X + 1).\ *.* the feature is marked as deprecated
+-  In PTB (X + 2).0.0 the feature is removed or changed
```

Comment:
this is confusing to me. We should just say that deprecated functionality will be removed after two major versions, giving an example say, `If a non-bot api feature is marked deprecated at say, 20.2, then we will remove it in '22.0'`.

Reply:
Btw, does #3632 violate this policy?

Reply:
I had hoped that this more schematic overview would make the policy more clear :D e.g. with "after two major versions" could also be interpreted as "if deprecated in 20.2, then it will be removed it 23.0 at earliest", (21.0 being the first and 22.0 being the second major version). We can still add an example, though

#3632 does not break the policy IMO since AFAIK we have documented nowhere that `Massage.date` & co are UTC-localized.